clusterman/mesos/util.py

```diff
@@ -103,7 +103,7 @@ def mesos_post(url: str, endpoint: str) -> requests.Response:
                 f'Response Text: {response.text}\n'
             )
         logger.critical(log_message)
-        raise PoolManagerError(f'Mesos master unreachable: check the logs for details') from e
+        raise PoolConnectionError(f'Mesos master unreachable: check the logs for details') from e
```

Comment:
I don't think it does. There isn't a single unifying exception that is raised when we fail to reach Mesos, so the point of PoolConnectionError here is so that we can catch errors from mesos_post in _autoscale.